gradient_boosting.py

```diff
@@ -0,0 +1,101 @@
+import numpy as np
+
+from sklearn.base import RegressorMixin
+from tree import DecisionTreeRegressor
+
+
+class BaseGradientBoostingRegressor:
+
+    def __init__(self,
+                 n_estimators: 5,
+                 max_depth: int,
+                 learning_rate=None):
+
+        self.estimator = DecisionTreeRegressor
+        self.n_estimators = n_estimators
+        self.max_depth = max_depth
+        self.learning_rate = [learning_rate] * n_estimators
+
+        self._estimator_params = self.estimator().get_params()
+
+        self._set_estimators()
+
+    def _set_estimators(self):
+        est_params = self._estimator_params
+
+        self._estimators = [self.estimator(**est_params)
+                            for _ in range(self.n_estimators)]
+
+    def fit(self, X, y):
+
+        y_pred = np.zeros_like(y)
+        self._init_pred = np.mean(y)
+        y_pred.fill(self._init_pred)
+        y_true = y
+
+        for i, est in enumerate(self._estimators):
+            y_err = y_true - y_pred
+            est.fit(X, y_err)
+            yi_pred = est.predict(X)
+            y_pred += self.learning_rate[i] * yi_pred
+
+    def predict(self, X):
+        y_pred = np.zeros(shape=(X.shape[0], 1)) + self._init_pred
+        for i, est in enumerate(self._estimators):
+            y_pred += self.learning_rate[i] * est.predict(X)
+        return y_pred
+
+
+class GradientBoostingRegressor(BaseGradientBoostingRegressor, RegressorMixin):
```

Comment:
Note, that this class contains no specific logic. It just adds the interface of regressor mixin. Please, remove it, and use just base class (without word 'Base' in the name) also with regressor mixin interface.